zetaclient/orchestrator/orchestrator.go

```diff
@@ -518,3 +565,60 @@ func (oc *Orchestrator) ScheduleCctxBTC(
 		}
 	}
 }
+
+// runObserverSignerSync runs a blocking ticker that observes chain changes from zetacore
+// and optionally (de)provisions respective observers and signers.
+func (oc *Orchestrator) runObserverSignerSync(ctx context.Context) error {
+	const cadence = 5 * time.Second
+
+	ticker := time.NewTicker(cadence)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-oc.stop:
+			oc.logger.Warn().Msg("runObserverSignerSync: stopped")
+			return nil
+		case <-ticker.C:
+			if err := oc.syncObserverSigner(ctx); err != nil {
+				oc.logger.Error().Err(err).Msg("runObserverSignerSync: syncObserverSigner failed")
+			}
+		}
+	}
+}
+
+// syncs and provisions observers & signers.
+// Note that zctx.AppContext Update is a responsibility of another component
+// See zetacore.Client{}.UpdateZetacoreContextWorker
+func (oc *Orchestrator) syncObserverSigner(ctx context.Context) error {
+	oc.mu.Lock()
+	defer oc.mu.Unlock()
+
+	client := oc.zetacoreClient
+
+	added, removed, err := syncObserverMap(ctx, client, oc.tss, oc.dbDirectory, oc.baseLogger, oc.ts, &oc.observerMap)
+	if err != nil {
+		return errors.Wrap(err, "syncObserverMap failed")
+	}
+
+	if added+removed > 0 {
+		oc.logger.Info().
+			Int("observer.added", added).
+			Int("observer.removed", removed).
+			Msg("synced observers")
+	}
+
+	added, removed, err = syncSignerMap(ctx, oc.tss, oc.baseLogger, oc.ts, &oc.signerMap)
+	if err != nil {
+		return errors.Wrap(err, "syncSignerMap failed")
+	}
+
+	if added+removed > 0 {
```

Comment:
Are newly created or newly deleted observer and signer pair always exactly match?

Reply:
That's a good question. We can continue the discussion here https://github.com/zeta-chain/node/issues/2519